[jacksonyuan256] ip

jacksonyuan256 opened the pull request:
#DukeBot
*Revolutionize your task management!*

DukeBot keep tracks of your tasks for you. It is
1. text based
2. easy to learn
3. ~~FAST~~ REALLY FAST

Instructions how to format input:
- todo **insert task here**
- deadline **insert task here** /by 2020-12-08 1800
- event **insert task here** /from Mon 2pm /to 4pm
- expense **insert expense**/$100
- editExpense **index no. of expense** $200

Other useful commands:
- **list** returns the current list of tasks
- **bye** exits the program
- **mark 1** marks the task as completed
- **unmark 1** unmarks the task as incomplete
- **report** returns the full expense report

Features:
- [x] Manage tasks
- [x] Manage deadlines
- [x] Manage events
- [x] Manage expenses

jacksonyuan256 committed:
Rename Greet Exit

jacksonyuan256 committed:
Echo

jacksonyuan256 committed:
Add, List

jacksonyuan256 committed:
Mark as Done

jacksonyuan256 committed:
ToDos, Events, Deadlines

jacksonyuan256 committed:
Automated Text UI Testing

jacksonyuan256 committed:
Handle Errors

jacksonyuan256 committed:
Delete

jacksonyuan256 committed:
Added the feature to save data

jacksonyuan256 committed:
Added features to rewrite dates in a different format

jacksonyuan256 committed:
Revert "Added features to rewrite dates in a different format"

This reverts commit 1e98197b68c3dc35eccaea4ee866368d9802c1e5.

jacksonyuan256 committed:
Added features to rewrite date/time

ran into some issues earlier and had to revert

jacksonyuan256 committed:
Merge Level-8

jacksonyuan256 committed:
Refactored code to extract code as classes

jacksonyuan256 committed:
Merge branch 'A-MoreOOP'

jacksonyuan256 committed:
Organise into packages

jacksonyuan256 committed:
Added gradle support

jacksonyuan256 committed:
Added JUnit tests

jacksonyuan256 committed:
Added JavaDocs

jacksonyuan256 committed:
Tweaked code according to coding standard

jacksonyuan256 committed:
Added the feature to find tasks via a keyword

jacksonyuan256 committed:
Merge branch 'branch-A-JavaDoc'

jacksonyuan256 committed:
Resolved merge conflicts

jacksonyuan256 committed:
Merge branch 'branch-Level-9'

jacksonyuan256 committed:
Some minor edits

jacksonyuan256 committed:
Added checkstyle to gradle

Minor edits to everything

jacksonyuan256 committed:
Merge branch 'branch-checkStyle'

markgcera commented on src/main/java/chatbot/UI.java:
```diff
@@ -0,0 +1,115 @@
+     * 
+     *  @param str The string to be printed.
+     */
+    public void incompleteCommand(String str) {
```
Perhaps the name of this method can include the verb "print"? Just for a little bit more clarity.

markgcera commented on src/main/java/chatbot/Task.java:
```diff
@@ -0,0 +1,60 @@
+    /**
+     * Marks the task as not completed
+     */
+    public void markTaskNotDone() {
```
I like the naming of all the methods so far, they are very clear, great job!

jacksonyuan256 committed:
Added GUI

jacksonyuan256 committed:
Merge branch 'branch-Level-10'

yihfei commented on src/main/java/chatbot/Duke.java:
```diff
@@ -0,0 +1,376 @@
+    }
+    
+    public static void main(String[] args) {
+        // new Duke().run();
```
Is the commented out code still necessary?

yihfei commented on src/main/java/chatbot/Parser.java:
```diff
@@ -0,0 +1,171 @@
+     * @param input The user input to be interpreted.
+     * @return The value to determine if the program ends. 
+     */
+    public String runInput(String input) {
```
Code here for this method is slightly long. Perhaps could consider implementing a command class?

yihfei commented on src/test/java/chatbot/DukeTest.java:
```diff
@@ -0,0 +1,65 @@
+public class DukeTest {
+
+    @Test
+    public void addDeleteTasksTest() {
```
I like the naming in this class, it's very informative and detailed!

yihfei commented on src/main/java/chatbot/Parser.java:
```diff
@@ -0,0 +1,171 @@
+                int index = taskHandler.getSize();
+
+                return ui.addTask(event, index);
+            }
```
Nonetheless, I still like how the method is still readable despite its long code

yihfei reviewed (commented):
Overall, generally clean code. I noticed quite a few commented out code but they don't pose any issues except for taking up lines in the code.

jacksonyuan256 committed:
Added java assertions in some classes

An assertion was added at the end of runInput method (Parser) since it
is not intended for the method to reach the end. One was also added for
retrieveTask method (TaskList) as it cannot be called if the length
of the arrayList is 0. For mark and unmark, we also want to ensure that
the correct completion status is printed. To do this, let's add an
assertion to check the boolean isDone.

jacksonyuan256 committed:
Edited some code to improve the code quality

By using the gradle checkstyle to detect style violations, most of it
has been rectified to the best of my means. I am still unsure why some
whitespace and newline errors still show up.

jacksonyuan256 committed:
Merge pull request #1 from jacksonyuan256/branch-A-CodeQuality

Edited some code to improve the code quality

jacksonyuan256 committed:
Merge pull request #2 from jacksonyuan256/branch-A-Assertions

Added java assertions in some classes

jacksonyuan256 committed:
Added an extension to record expenses

This new feature allows users to record their expenses, to edit the
costs and subsequently to see the full report.

jacksonyuan256 committed:
Merge branch 'branch-Extension'

jacksonyuan256 committed:
Some code quality changes

jacksonyuan256 committed:
Added some personality to the bot and code quality changes.

jacksonyuan256 committed:
Merge branch 'branch-A-Personality'

jacksonyuan256 committed:
Created a brief User Guide

jacksonyuan256 committed:
Added the Ui.png in the docs folder

jacksonyuan256 committed:
Some minor changes to build.gradle so the .jar file can be built

In addition, some edits were made to better handle incomplete inputs.

jacksonyuan256 committed:
Major bug fixes with loading and writing files

jacksonyuan256 committed:
Update README.md
